src/renderers/dom/shared/eventPlugins/ChangeEventPlugin.js

```diff
@@ -148,33 +272,34 @@ function handleControlledInputBlur(inst, node) {
 var ChangeEventPlugin = {
   eventTypes: eventTypes,
 
-  extractEvents(topLevelType, targetInst, nativeEvent, nativeEventTarget) {
+  _isInputEventSupported: isInputEventSupported,
+
+  extractEvents: function(
+    topLevelType,
+    targetInst,
+    nativeEvent,
+    nativeEventTarget,
+  ) {
     var targetNode = targetInst
       ? ReactDOMComponentTree.getNodeFromInstance(targetInst)
       : window;
 
-    // On the selectionchange event, the target is the document which isn't
-    // helpful becasue we need the input, so we use the activeElement instead.
-    if (!isTextInputEventSupported && topLevelType === 'topSelectionChange') {
-      nativeEventTarget = targetNode = getActiveElement();
-
-      if (targetNode) {
-        targetInst = ReactDOMComponentTree.getInstanceFromNode(targetNode);
-      }
-    }
-
     var getTargetInstFunc, handleEventFunc;
-
     if (shouldUseChangeEvent(targetNode)) {
       getTargetInstFunc = getTargetInstForChangeEvent;
-    } else if (isTextInputElement(targetNode) && !isTextInputEventSupported) {
-      getTargetInstFunc = getTargetInstForInputEventPolyfill;
-    } else {
-      getTargetInstFunc = getTargetInstForInputOrChangeEvent;
+    } else if (isTextInputElement(targetNode)) {
+      if (isInputEventSupported) {
+        getTargetInstFunc = getTargetInstForInputOrChangeEvent;
+      } else {
+        getTargetInstFunc = getTargetInstForInputEventPolyfill;
+        handleEventFunc = handleEventsForInputEventPolyfill;
+      }
+    } else if (shouldUseClickEvent(targetNode)) {
```

Comment:
it'd be nice to keep the change removing the click behavior for checkable inputs. I can see if I can just pull those bits out. I guess we have no reason to think this isn't the cause here.